### PR TITLE
Feature/tao 2740 2741 tei scoring

### PR DESCRIPTION
**Pull request by ssipasseuth**
requires https://github.com/oat-parcc/extension-tao-itemqti/pull/11
requires https://github.com/oat-parcc/extension-tao-itemqti/pull/10

### CODE REVIEW

File: views/js/pciCreator/dev/fractionModelInteraction/creator/widget/states/Answer.js [hunk 1]
```diff
@@ -0,0 +1,64 @@
+define([
+    'lodash',
+    'jquery',
+    'i18n',
+    'taoQtiItem/qtiCreator/widgets/states/factory',
+    'taoQtiItem/qtiCreator/widgets/interactions/states/Answer',
+    'taoQtiItem/qtiCreator/widgets/helpers/formElement',
+    'taoQtiItem/qtiCreator/widgets/interactions/helpers/answerState'
```
Comment (no-chris) [inline views/js/pciCreator/dev/fractionModelInteraction/creator/widget/states/Answer.js]: it seems these imports are not used

Comment (ssipasseuth) [inline views/js/pciCreator/dev/fractionModelInteraction/creator/widget/states/Answer.js]: right !

File: views/js/pciCreator/dev/graphFunctionInteraction/creator/tpl/propertiesForm.tpl [hunk 2]
```diff
@@ -3,49 +3,170 @@
...
+        <input name="xLabel" value="{{xLabel}}" type="text" />
+    </div>
+    <div class="panel creator-graphPointLineGraphInteraction-spinner">
+        <label for="xSubStep" class="spinner">{{__ "sub step (snapping)"}}</label>
+        <input name="xSubStep" value="{{xSubStep}}" data-increment="1" data-min="1" type="text" />
+    </div>
+    <label>
+        <input name="xAllowOuter" type="checkbox" {{#if xAllowOuter}}checked="checked"{{/if}}/>
+        <span class="icon-checkbox"></span>
+        {{__ "allow points on border"}}
```
Comment (no-chris) [inline views/js/pciCreator/dev/graphFunctionInteraction/creator/tpl/propertiesForm.tpl:67]: this behaviour doesn't seems implemented ?

File: views/js/pciCreator/dev/graphFunctionInteraction/creator/widget/states/Question.js [hunk 3]
```diff
@@ -44,34 +44,69 @@ define([
...
-            yMax : interaction.prop('yMax'),
-            graphColor : interaction.prop('graphColor'),
-            graphWidth : interaction.prop('graphWidth')
+
+            graphTitle : interaction.prop('graphTitle'),
+            graphTitleRequired : getBoolean(interaction.prop('graphTitleRequired'), true),
+
+            maxPoints : interaction.prop('maxPoints'),
+            draggable : getBoolean(interaction.prop('draggable'), true),
+            segment : getBoolean(interaction.prop('segment'), true),
```
Comment (no-chris) [inline views/js/pciCreator/dev/graphFunctionInteraction/creator/widget/states/Question.js]: those should be removed ?

File: views/js/pciCreator/dev/graphFunctionInteraction/pciCreator.js [hunk 4]
```diff
@@ -34,12 +36,38 @@ define([
...
             return {
                 graphs : 'linear,absolute,quadratic,exponential,logarithmic,cosine,tangent',
-                xMin : -10,
-                xMax : 10,
-                yMin : -10,
-                yMax : 10,
-                graphColor : '#bb1a2a',
-                graphWidth : 3
+
+                "draggable": true,
```
Comment (no-chris) [inline views/js/pciCreator/dev/graphFunctionInteraction/pciCreator.js]: also outdated ?

File: views/js/pciCreator/dev/graphFunctionInteraction/runtime/graphFunctionInteraction.js [hunk 5]
```diff
@@ -20,46 +20,127 @@ define([
...
+            yWeight = parseInt(rawConfig.yWeight, radix),
+            yBorderWeight = parseInt(rawConfig.yBorderWeight, radix),
+
+            gridConfig = {
+                // PCI config
+                draggable: getBoolean(rawConfig.draggable, true),
+                graphType: rawConfig.graphType, // scatterPlot or line
+                //maxPoints: parseInt(rawConfig.maxPoints, radix),
+                segment: getBoolean(rawConfig.segment, true), // draw only segments between points
+
```
Comment (no-chris) [inline views/js/pciCreator/dev/graphFunctionInteraction/runtime/graphFunctionInteraction.js:42]: outdated and commented values ?

File: views/js/pciCreator/dev/graphFunctionInteraction/runtime/graphFunctionInteraction.js [hunk 6]
```diff
@@ -121,8 +202,8 @@ define([
                             fx = grid.getX() + Math.round((event.clientX - bnds.left) / bnds.width * grid.getWidth() * wfactor),
                             fy = grid.getY() + Math.round((event.clientY - bnds.top) / bnds.height * grid.getHeight() * wfactor);
 
-                        // Create the first point or the second or replace the second according the rules defined by the client                  
-                        if(points.length < 2){
+                        // Create the first point or the second or replace the second according the rules defined by the client
+                        if(points.length < 2 && areCoordsValid(fx, fy)){
```
Comment (no-chris) [inline views/js/pciCreator/dev/graphFunctionInteraction/runtime/graphFunctionInteraction.js]: line 216 
throw TypeError : oldPoint is not defined

to reproduce:
- create new interaction
- remove all points
- uncheck allow on border
- click on the border

Also, after a click on the border, if you add another point anywhere else, then you are then able to add a point on the border.

wrapping this whole if then else in areCoordsValid **MIGHT** fix those problems. See : 
https://github.com/oat-sa/extension-parcc-tei/blob/develop/views/js/pciCreator/dev/graphPointLineGraphInteraction/runtime/graphPointLineGraphInteraction.js#L175

File: views/js/test/fraction/test.js [hunk 7]
```diff
@@ -78,9 +78,9 @@ define([
```
Comment (no-chris) [inline views/js/test/fraction/test.js:78]: this test stops after the first test on my machine ?

File: views/js/pciCreator/dev/graphFunctionInteraction/creator/tpl/propertiesForm.tpl [hunk 8]
```diff
@@ -3,49 +3,146 @@
...
-    <label for="yMax" class="spinner">to</label>
-    <input name="yMax" value="{{yMax}}" data-increment="1" type="text" />
+    <br />
+    <label>{{__ "Graduation"}}</label>
+    <div class="panel subpanel">
+        <div class="panel creator-graphPointLineGraphInteraction-range">
+            <label for="yytart" class="spinner">{{__ "from"}}</label>
+            <input name="yStart" value="{{yStart}}" data-increment="1" type="text" />
+            <label for="yEnd" class="spinner">{{__ "to"}}</label>
+            <input name="xynd" value="{{yEnd}}" data-increment="1" type="text" /><br />
```
Comment (no-chris) [inline views/js/pciCreator/dev/graphFunctionInteraction/creator/tpl/propertiesForm.tpl]: ups...
that one is probably in Point Graph too...

File: views/js/test/fraction/test.js (hunk 7 above)
Comment (ssipasseuth) [inline views/js/test/fraction/test.js:78]: you are right, sorry, this PR also requires another fix : https://github.com/oat-parcc/extension-tao-itemqti/pull/11

**Pull request merged**